git_machete/client.py

```diff
@@ -946,7 +946,7 @@ def delete_unmanaged(self) -> None:
                 opt_yes_msg = f"Deleting branch {msg_core}"
                 ans = self.ask_if(msg, opt_yes_msg)
                 if ans in ('y', 'yes'):
-                    self.__git.run_git("branch", "-d" if is_merged_to_remote else "-D", branch)
+                    self.__git.run_branch("-d" if is_merged_to_remote else "-D", branch)
```

Comment:
Maybe `delete_branch`? with a boolean to param of some adequate name to choose b/w `-D`/`-d` (the flag itself should be an implementation detail, should not leak into the API of `GitContext`)